### ISSUE TITLE
[rush] include `tags` field to the output of `rush list --json`

### ISSUE THREAD

**Issue by lukaskl** [microsoft/rushstack#3399]
## Summary

This is a follow up of https://github.com/microsoft/rushstack/pull/3300 and slightly https://github.com/microsoft/rushstack/issues/3355

Right now when we execute `rush list --json` `tags` field is not included in the outputted json, but it would be helpful and more convenient if it would be. 

## Repro steps

Let's say that I have these projects inside my `rush.json`

```jsonc
{
// ...
"projects": [
    {
      "packageName": "cloud-frontend",
      "projectFolder": "packages/cloud-frontend",
      "tags": ["libraries", "web"]
    },
    {
      "packageName": "qa-automation",
      "projectFolder": "packages/qa-automation",
      "tags": ["test"]
    }
  ]
}
```

when I execute `rush list --json`:

  **Expected result:** 

```diff
{
  "projects": [
    {
      "name": "cloud-frontend",
      "version": "0.0.0",
      "path": "packages/cloud-frontend",
      "fullPath": "<...>/packages/cloud-frontend",
      "shouldPublish": false,
+    "tags": ["libraries", "web"]
    },
    {
      "name": "qa-automation",
      "path": "packages/qa-automation",
      "fullPath": "<...>/packages/qa-automation",
      "shouldPublish": false,
+    "tags": ["test"]
    }
  ]
}
```
  **Actual result:** 

```json
{
  "projects": [
    {
      "name": "cloud-frontend",
      "version": "0.0.0",
      "path": "packages/cloud-frontend",
      "fullPath": "<...>/packages/cloud-frontend",
      "shouldPublish": false
    },
    {
      "name": "qa-automation",
      "path": "packages/qa-automation",
      "fullPath": "<...>/packages/qa-automation",
      "shouldPublish": false
    }
  ]
}
```

## Details

In addition, I can confirm that `tags` are working in the first place, e.g. if I execute `rush list --only tag:web`, the output is:

```
Rush Multi-Project Build Tool 5.68.1 - https://rushjs.io
Node.js version is 16.13.1 (LTS)


Starting "rush list"

cloud-frontend
```

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | `5.68.1` |
| `rushVersion` from rush.json? | `5.68.1` |
| `useWorkspaces` from rush.json? | `true` |
| Operating system? |  Mac |
| Would you consider contributing a PR? | if someone would guide me - I would |
| Node.js version (`node -v`)? | `16.13.1` |

**Comment by lukaskl** [comment]
Also `rush list --detailed` outputs:

```
Rush Multi-Project Build Tool 5.68.1 - https://rushjs.io
Node.js version is 16.13.1 (LTS)


Starting "rush list"


ERROR: Cannot read properties of undefined (reading 'toString')
```

Not sure is it worthy to create a dedicated ticket for this or not 🤔

**Comment by iclanton** [comment]
@lukaskl - I'm not able to repro that issue. Can you run `rush --debug list --detailed` and share the stacktrace?

**Comment by lukaskl** [comment]
> @lukaskl - I'm not able to repro that issue. Can you run `rush --debug list --detailed` and share the stacktrace?

<details>
  <summary>rush --debug list --detailed</summary>

```
Rush Multi-Project Build Tool 5.68.1 - https://rushjs.io
Node.js version is 16.15.0 (LTS)


Starting "rush list"


ERROR: Cannot read properties of undefined (reading 'toString')

TypeError: Cannot read properties of undefined (reading 'toString')
    at /Users/lukasklusis/.nvm/versions/node/v18.1.0/lib/node_modules/@microsoft/rush/node_modules/cli-table/lib/index.js:188:27
    at Array.forEach (<anonymous>)
    at generateRow (/Users/lukasklusis/.nvm/versions/node/v18.1.0/lib/node_modules/@microsoft/rush/node_modules/cli-table/lib/index.js:187:11)
    at /Users/lukasklusis/.nvm/versions/node/v18.1.0/lib/node_modules/@microsoft/rush/node_modules/cli-table/lib/index.js:281:16
    at Table.forEach (<anonymous>)
    at Table.toString (/Users/lukasklusis/.nvm/versions/node/v18.1.0/lib/node_modules/@microsoft/rush/node_modules/cli-table/lib/index.js:264:10)
    at ListAction._printListTable (/Users/lukasklusis/.nvm/versions/node/v18.1.0/lib/node_modules/@microsoft/rush/node_modules/@microsoft/rush-lib/lib/cli/actions/ListAction.js:172:27)
    at ListAction.runAsync (/Users/lukasklusis/.nvm/versions/node/v18.1.0/lib/node_modules/@microsoft/rush/node_modules/@microsoft/rush-lib/lib/cli/actions/ListAction.js:71:18)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```
    
</details>

But I believe I understood what is going on.
One of my `package.json` didn't have `version` field (newbie mistake, I've just started migration to `rush`). 

I've added `"version": "0.0.0",` field to that `package.json` and now it is fixed. 

but adding more details, it seems that the exception was thrown by the `cli-table` package [here](https://github.com/Automattic/cli-table/blob/v0.3.11/lib/index.js#L188). And at that point my `items` values was:
```js
[
  'qa-automation',
  undefined,
  'packages/qa-automation',
  '',
  '',
  'false',
  ''
]
```

**Comment by iclanton** [comment]
Cool. Probably should have reported a better error, but I'm not sure exactly where we'd do that. What would you expect that error to look like?

**Comment by lukaskl** [comment]
Well, if I would have a saying, I would vote to simply handle this case, two reasons why: 

- it seems that `packageRow` at [ListAction.ts](https://github.com/microsoft/rushstack/blob/35a0bdc8a832b389456af2b9d985fda8975b4302/libraries/rush-lib/src/cli/actions/ListAction.ts#L201) is not expecting anything else than string items in the first place,
- and it seems that all other commands were working quite well. 

So maybe rush cli could just simply handle this case and output the table with a blank cell or `"undefined"`?
i.e. maybe it would be worthy just to add a safeguard.

Looking at your PR it seems like the same [`ListAction.ts`](https://github.com/microsoft/rushstack/blob/35a0bdc8a832b389456af2b9d985fda8975b4302/libraries/rush-lib/src/cli/actions/ListAction.ts) could be a good place to do so 🤔  either by making sure that every single item added to the `packageRow` is indeed a string, or adding a safeguard before [pushing the `packageRow` to the `table`](https://github.com/microsoft/rushstack/blob/35a0bdc8a832b389456af2b9d985fda8975b4302/libraries/rush-lib/src/cli/actions/ListAction.ts#L233), e.g.:


```diff
+    /**
+     * packageRow items should be only of string type, but other types still might end up here
+     * in this list, e.g. see: https://github.com/microsoft/rushstack/issues/3399#issuecomment-1121494993
+     * Therefore let's make sure that all items are indeed strings
+     */
+     const packageRowStrs = packageRow.map(cell => String(cell))
+     table.push(packageRowStrs);
-     table.push(packageRow);
```

If there is a wish, I could attempt to create this PR myself, just maybe not today anymore (it is already midnight at my end 😅)

**Comment by iclanton** [comment]
@lukaskl - This has been released with Rush v5.69.0.
